zio-schema-json/shared/src/main/scala/zio/schema/codec/JsonCodec.scala

```diff
@@ -991,7 +991,29 @@ object JsonCodec {
     import JsonCodec.JsonDecoder.schemaDecoder
 
     private[codec] def caseClass0Decoder[Z](discriminator: Int, schema: Schema.CaseClass0[Z]): ZJsonDecoder[Z] = { (trace: List[JsonError], in: RetractReader) =>
-      if (discriminator == -1) Codecs.unitDecoder.unsafeDecode(trace, in)
+      def skipField(): Unit = {
+        val rejectExtraFields = schema.annotations.collectFirst({ case _: rejectExtraFields => () }).isDefined
```

Comment:
Calculating this every time may seem redundant, but `skipField()` won't run at all in most cases.

Reply:
I still think we can make this a val on `Schema.Record`